api/src/paths/project/{projectId}/delete.ts

```diff
@@ -66,6 +67,35 @@ export function deleteProject(): RequestHandler {
     try {
       /**
        * PART 1
+       * Check that user is a system administrator - can delete a project (published or not)
+       * Check that user is a project administrator - can delete a project (unpublished only)
+       *
+       */
+
+      const getProjectSQLStatement = getProjectSQL(Number(req.params.projectId));
+
+      if (!getProjectSQLStatement) {
+        throw new HTTP400('Failed to build SQL get statement');
+      }
+
+      await connection.open();
+
+      const projectData = await connection.query(getProjectSQLStatement.text, getProjectSQLStatement.values);
+
+      await connection.commit();
+
+      const projectResult = (projectData && projectData.rows && projectData.rows[0]) || null;
+
+      if (!projectResult || !projectResult.id) {
+        throw new HTTP400('Failed to get the project');
+      }
+
+      if (req['system_user']['role_names'][0] === SYSTEM_ROLE.PROJECT_ADMIN && projectResult.publish_date) {
+        throw new HTTP400('Cannot delete a published project.');
+      }
```

Comment:
We'll need to tweak this a bit, because we can't assume that the 0th element is the right role, if they have multiple.  We can leverage that function in auth-utils:

```typescript
      if (userHasValidSystemRoles([SYSTEM_ROLE.PROJECT_ADMIN], [req['system_user']['role_names']]) && projectResult.publish_date) {
        throw new HTTP400('Cannot delete a published project.');
      }
```